pkg/fleetautoscalers/fleetautoscalers.go

```diff
@@ -41,100 +42,131 @@ var client = http.Client{
-	urlStr := ""
-	if w.URL != nil {
-		urlStr = *w.URL
+
+	if f == nil {
+		return 0, false, errors.New("nil Fleet passed")
 	}
-	var faResp autoscalingv1.FleetAutoscaleReview
-	servicePath := ""
-	if w.Service != nil {
-		if w.Service.Path != nil {
-			servicePath = *w.Service.Path
+
+	if w.URL != nil && w.Service != nil {
+		return 0, false, errors.New("service and url cannot be used simultaneously")
+	}
+
+	var u *url.URL
+	var err error
+
+	if w.URL != nil {
+		if *w.URL == "" {
+			return 0, false, errors.New("URL was not provided")
 		}
+		u, err = url.ParseRequestURI(*w.URL)
 		if err != nil {
-			return f.Status.Replicas, false, err
+			return 0, false, err
+		}
+	} else {
+		if w.Service.Name == "" {
+			return 0, false, errors.New("service name was not provided")
+		}
+
+		var servicePath string
+		if w.Service.Path != nil {
+			servicePath = *w.Service.Path
 		}
 
 		if w.Service.Namespace == "" {
 			w.Service.Namespace = "default"
 		}
-		scheme := "http://"
+
+		scheme := "http"
 		if w.CABundle != nil {
-			scheme = "https://"
+			scheme = "https"
+
+			// We can have multiple fleetautoscalers with different CABundles defined,
+			// so we switch client.Transport before each POST request
+			rootCAs := x509.NewCertPool()
+			if ok := rootCAs.AppendCertsFromPEM(w.CABundle); !ok {
+				return 0, false, errors.New("no certs were appended from caBundle")
+			}
+			client.Transport = &http.Transport{
+				TLSClientConfig: &tls.Config{
+					RootCAs: rootCAs,
+				},
+			}
 		}
-		urlStr = fmt.Sprintf("%s%s.%s.svc:8000/%s", scheme, w.Service.Name, w.Service.Namespace, servicePath)
+
+		u = buildURL(scheme, w.Service.Name, w.Service.Namespace, servicePath)
 	}
-	if urlStr == "" {
-		return f.Status.Replicas, false, errors.New("URL was not provided")
+
+	faReq := autoscalingv1.FleetAutoscaleReview{
+		Request: &autoscalingv1.FleetAutoscaleRequest{
+			UID:       uuid.NewUUID(),
+			Name:      f.Name,
+			Namespace: f.Namespace,
+			Status:    f.Status,
+		},
+		Response: nil,
 	}
 
-	u, err := url.Parse(urlStr)
+	b, err := json.Marshal(faReq)
 	if err != nil {
-		return f.Status.Replicas, false, err
+		return 0, false, err
 	}
 
-	// We could have multiple fleetautoscalers with different CABundles defined,
-	// so we switch client.Transport before each POST request
-	if u.Scheme == "https" {
-		rootCAs := x509.NewCertPool()
-		if ok := rootCAs.AppendCertsFromPEM(w.CABundle); !ok {
-			return f.Status.Replicas, false, errors.New("no certs were appended from caBundle")
-		}
-		tr := &http.Transport{
-			TLSClientConfig: &tls.Config{
-				RootCAs: rootCAs,
-			},
-		}
-		client.Transport = tr
-	}
 	res, err := client.Post(
-		urlStr,
+		u.String(),
 		"application/json",
 		strings.NewReader(string(b)),
 	)
 	if err != nil {
-		return f.Status.Replicas, false, err
+		return 0, false, err
 	}
-	defer res.Body.Close() // nolint: errcheck
+	defer func() {
+		if cerr := res.Body.Close(); cerr != nil {
+			log.Error(cerr)
```

Comment:
I think we can use `c.baseLogger` here, we could add receiver to the function `(c *Controller) applyWebhookPolicy()`:
https://github.com/googleforgames/agones/blob/cf3d21dbd4d2200760f0edf0198dfd4de2806477/pkg/fleetautoscalers/controller.go#L87

Reply:
Decided to handle it in a different way, please check.